client/src/utils.ts

```diff
@@ -30,18 +31,34 @@ const catchErr = (err: any) => {
 };
 
 // fetch response from url
-const fetchRes = async (reqType: string, data: any, url: string) => {
+// fetch response from url
+const fetchRes = async (
+  reqType: string,
+  data: Object,
+  url: string,
+  auth: Boolean = false
+) => {
   if (data == null) {
     return await fetch(url);
   } else {
+    const headers = new Headers({
+      "Content-Type": "application/json",
+    });
+    if (auth && !isWeb) {
+      const nonce = await fetchText("GET", {}, "/api/nonce");
+      const request = { nonce: nonce };
+      const idToken = microapps.getIdentity(request).catch((err: any) => {
+        return null;
+      });
+      headers.append("Authorization", "Bearer " + idToken);
```

Comment:
Extract the logic of "Bearer " + idToken into some common library, and that library should provide a function to generate the header with "Bearer" and provide a function to get the idToken from the header.

Otherwise, this becomes hard coded logic in client and server and there is no guarantee to make sure they are consistent.

Reply:
That makes sense. But the issue is that generating the header happens client-side, while getting the idToken from the header happens server-side, so the functions will have to be in separate files?

Reply:
In my Opinion, it should be a single file/class.
It would not make sense to split it into two files.
Consider you are defining a protocol, that the header should be "Bearer" + idToken, you should come up with a library which encapsulate all the related functions to this protocol.
Image you are writing an application which has a sender and receiver, and sender would send out message with RSA encryption, the receiver would use RSA decryption, you should import the same RSA library for both sender and receiver to make sure the consistency.

Reply:
https://github.com/googleinterns/scan-and-go/issues/173 Created a issue for you. I would just approve this MR first. It is up to you whether you want to do the refactoring here or in following MR given your current MR is already very big.